### PR TITLE
Increase deploy timeouts and make them configurable

### PR DESCRIPTION
**Pull request by aguerra**
Also upgrade to the latest envconfig.

---
This change is [Reviewable](https://reviewable.io/reviews/luizalabs/teresa-api/170)

### CODE REVIEW

File: k8s/deployments.go [hunk 1]
```diff
@@ -59,13 +59,15 @@ type deploy struct {
...
+	FinishTimeout        time.Duration `split_words:"true" default:"30m"`
+	RevisionHistoryLimit int32         `split_words:"true" default:"5"`
+	StartTimeout         time.Duration `split_words:"true" default:"10m"`
 }
 
 var deploymentConfig DeploymentConfig
 
 func init() {
-	err := envconfig.Process("teresadeploy", &deploymentConfig)
+	err := envconfig.Process("teresa_deploy", &deploymentConfig)
```
Comment (drgarcia1986) [inline k8s/deployments.go]: Why?
Is it because the upgrade of `envconfig` dependency ?

**Comment by drgarcia1986** [comment]
Related #139 #151

File: k8s/deployments.go (hunk 1 above)
Comment (aguerra) [inline k8s/deployments.go]: No, I prefer env vars `TERESA_* style.

Comment (drgarcia1986) [inline k8s/deployments.go]: Hum 🤔 but, this change will broke the env `TERESADEPLOY_REVISION_HISTORY_LIMIT` and is out of pattern of others env configs.

What do you think about?

Comment (aguerra) [inline k8s/deployments.go]: I would changed it right now as it is undocumented, but to be on the safer side we can change the env vars style for `0.3.0` and document the change. Agree?

Comment (drgarcia1986) [inline k8s/deployments.go]: It's "documented" on README...

But :shipit:

**Pull request merged**